src/hotspot/share/utilities/tribool.hpp

```diff
@@ -40,11 +40,14 @@ class TriBool{
 
  public:
   TriBool() : _value(0) {}
-  TriBool(bool value) : _value(((u1)value) | 2) {}
+  TriBool(bool value) : _value(value) {
+    _value = _value | 2; // set to not-default
```

Comment:
Why not `|=` here?

Reply:
I was trying different versions to see if |= would give a warning.